learnersync.go

```diff
@@ -418,25 +418,29 @@ func (s *Sync) readAttendanceIdFile() (string, error) {
 }
 
 func (s *Sync) WaitForAttendanceId() error {
+	var httpEnabled = !s.InHostedEnv
+	var err error = nil
```

Comment:
Was there previously a bug because this line wasn't here? I can't work out from reading this code where the `err` on line 463 was coming from before.

Reply:
Ah - got it. Previously `err` was always being initialised by line 423, and updated by line 450 - now it's not always initialised on 423.

This feels like quite confusing code structure! Would definitely have written some of this differently in the first place (no need to fix now).